File: 백준/Silver/35424. 낙서 제거/낙서 제거.py
import sys
input = sys.stdin.readline

N, K = map(int, input().split())
a = list(map(int, input().split()))

i = 0
answer = 0

while i < N:
    j = i
    min_val = a[i]
    max_val = a[i]

    while j < N:
        min_val = min(min_val, a[j])
        max_val = max(max_val, a[j])

        area = (j - i + 1) * (max_val - min_val + 1)

        if area > K:
            break

        j += 1

    answer += 1
    i = j

print(answer)

File: 백준/Silver/35424. 낙서 제거/낙서 제거.js
const readline=require('readline')
const rl=readline.createInterface({
  input:process.stdin,
  output:process.stdout
})

const input=[]
let iter=0
rl.on('line',(line)=>{
  input.push(line)
}).on('close',()=>{
  const [N,K]=input[iter++].split(' ').map(item=>Number(item))
  const l=input[iter++].split(' ').map(item=>Number(item)-1)

  let line=0
  let start=l[0]
  let end=l[0]

  let answer=0
  for(let i=0;i<N;i++){
    start=Math.min(start,l[i])
    end=Math.max(end,l[i])
    let now=(i-line+1)*(end-start+1)
    if (now<=K) continue
    line=i
    answer+=1
    start=l[i]
    end=l[i]
  }

  console.log(answer+1)
  process.exit()
})
